airflow/contrib/operators/aws_lambda_operator.py

```diff
@@ -0,0 +1,108 @@
+
+
+class AwsLambdaInvokeFunctionOperator(BaseOperator):
+    """
+    Invoke AWS Lambda functions with a JSON payload.
+
+    The check_success_function signature should be a single param which will receive a dict.
+    The dict will be the "Response Structure" described in http://bit.ly/2KXLWWF. It may
+    be necessary to read the Payload see the actual response from the Lambda function e.g.,
+
+    ```
+    def succeeded(response):
+        payload = json.loads(response['Payload'].read())
+        # do something with payload
+    ```
+
+    :param function_name: The name of the Lambda function.
+    :type function_name: str
+    :param region_name: AWS region e.g., eu-west-1, ap-southeast-1, etc.
+    :type region_name: str
+    :param payload: The JSON to submit as input to a Lambda function.
+    :type payload: str
+    :param check_success_function: A function to check the Lambda response and determine success or failure.
+    :type check_success_function: function
+    :param log_type: Tail Invocation Request
+    :type log_type: str
+    :param qualifer: A version or alias name for the Lambda.
+    :type qualifer: str
+    :param invocation_type: One of "Event" | "RequestResponse" | "DryRun". Typically "RequestResponse"
+    :type invocation_type: str
+    :param aws_conn_id: connection id of AWS credentials / region name. If None,
+        credential boto3 strategy will be used
+        (http://boto3.readthedocs.io/en/latest/guide/configuration.html).
+    :type aws_conn_id: str
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        function_name,
+        region_name,
+        payload,
+        check_success_function,
+        log_type="None",
+        qualifier="$LATEST",
+        invocation_type="RequestResponse",
+        aws_conn_id=None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.function_name = function_name
+        self.region_name = region_name
+        self.payload = payload
+        self.check_success_function = check_success_function
+        self.invocation_type = invocation_type
+        self.log_type = log_type
+        self.qualifier = qualifier
+        self.aws_conn_id = aws_conn_id
+
+    @property
+    def aws_lambda_hook(self):
+        if not hasattr(self, "_aws_lambda_hook"):
+            self._aws_lambda_hook = AwsLambdaHook(
+                self.function_name,
+                self.region_name,
+                self.log_type,
+                self.qualifier,
+                self.invocation_type,
+                aws_conn_id=self.aws_conn_id
+            )
+        return self._aws_lambda_hook
+
+    def execute(self, context):
+        self.log.info(f"AWS Lambda: invoking {self.function_name}")
+
+        response = self.aws_lambda_hook.invoke_lambda(self.payload)
+        if not self._succeeded(response):
+            self.log.error(response)
+            raise Exception(f"AWS Lambda: {self.function_name} failed!")
```

Comment:
Can you narrow down the exception type? So that you can better `catch` (and test) the exception being raised.